applications/console_text_messenger/src/main.rs

```diff
@@ -214,8 +217,29 @@ pub fn main() {
     let secret_key = CommsSecretKey::from_hex(settings.secret_key.unwrap().as_str()).unwrap();
     let public_key = CommsPublicKey::from_secret_key(&secret_key);
 
-    // TODO Use a less hacky crate to determine the local machines public IP address. This only works on Unix systems!
-    let ip = local_ip::get().expect("Could not determine local machines public IP address");
+    let mut address_count = 0;
+    let mut ip = String::new();
+    // loop through network interfaces
+    let interfaces = datalink::interfaces();
+    for interface in interfaces.into_iter() {
```

Comment:
Rather than a for loop, you may want to use something like: (didn't test this)

```rust
let local_ip = interfaces.into_iter()
.filter(|interface| !interface.is_loopback())
.find_map(|interface| interface.ips.iter().find(|addr| if addr.is_ipv4() { Some(addr.ip().to_string() } else { None }))
```
This will find the first non-loopback ipv4 address (rather than the last).
I suggest putting this into a function (eg. `get_local_ipv4_address`) which will make things clearer.

The `address_count == 0` check below would become `local_ip.is_none()`

Reply:
I didn't get that to work like the above, though I did get close.

Filtered it down to interfaces that contained an ipv4 address, selected the first one (checking if it not None) and then looped through the two addresses in it, one being the ipv4 address.